### PR TITLE
Expand PriorityQueue test coverage

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Extends PriorityQueue test coverage in the following ways:

* Abstracts the `IComparer<TPriority>` parameter in the PriorityQueue generic test class so that all code paths added in #48346 are exercised.
* Adds a few missing unit tests.
* Minor refactorings and rearrangements of existing tests.
* Adds property tests for PriorityQueue, a form of structured fuzzing already used in the Cbor project.

### CODE REVIEW

File: src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs [hunk 1]
```diff
@@ -41,9 +39,9 @@ protected IEnumerable<(TElement, TPriority)> GenericIEnumerableFactory(int count
         #region Constructors
 
         [Fact]
-        public void PriorityQueue_Generic_Constructor()
+        public void PriorityQueue_DefaultConstructor_ComparerEqualsDefaultComparer()
         {
-            var queue = new PriorityQueue<TElement, TPriority>();
+            var queue = new PriorityQueue<TPriority, TPriority>();
```
Comment (stephentoub) [inline src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs:44]: Was this a purposeful change?

File: src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs [hunk 2]
```diff
@@ -52,124 +50,100 @@ public void PriorityQueue_Generic_Constructor()
...
         [MemberData(nameof(ValidCollectionSizes))]
-        public void PriorityQueue_Generic_Constructor_int_IComparer(int initialCapacity)
+        public void PriorityQueue_CapacityConstructor_ComparerShouldEqualDefaultComparer(int initialCapacity)
         {
-            IComparer<TPriority> comparer = Comparer<TPriority>.Default;
             var queue = new PriorityQueue<TElement, TPriority>(initialCapacity);
-
             Assert.Empty(queue.UnorderedItems);
-            Assert.Equal(comparer, queue.Comparer);
+            Assert.Equal(Comparer<TPriority>.Default, queue.Comparer);
```
Comment (stephentoub) [inline src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs:81]: It doesn't really matter, but consider using Assert.Same when you're looking to validate reference equality.

File: src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs [hunk 3]
```diff
@@ -52,124 +50,100 @@ public void PriorityQueue_Generic_Constructor()
...
         }
 
         [Theory]
         [MemberData(nameof(ValidCollectionSizes))]
-        public void PriorityQueue_Generic_Constructor_IEnumerable(int count)
+        public void PriorityQueue_EnumerableConstructor_ShouldContainAllElements(int count)
         {
-            HashSet<(TElement, TPriority)> itemsToEnqueue = this.GenericIEnumerableFactory(count).ToHashSet();
+            var itemsToEnqueue = CreateItems(count).ToArray();
```
Comment (stephentoub) [inline src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs:88]: Nit: var => whatever type this is

File: src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+namespace System.Collections.Tests
+{
+    public static class PriorityQueue_PropertyTests
+    {
+        const int MaxTest = 100;
+
+        private readonly static IComparer<string> s_stringComparer = StringComparer.Ordinal;
+        
+        [Property(MaxTest = MaxTest)]
```
Comment (stephentoub) [inline src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs:18]: I haven't seen this construct in our tests before.  What is this doing?

File: src/libraries/System.Collections/tests/System.Collections.Tests.csproj [hunk 5]
```diff
@@ -129,4 +132,8 @@
     <Compile Include="$(CommonTestPath)System\Collections\IDictionary.NonGeneric.Tests.netcoreapp.cs"
              Link="Common\System\Collections\IDictionary.NonGeneric.Tests.netcoreapp.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="FsCheck" Version="$(FsCheckVersion)" />
+    <PackageReference Include="FsCheck.Xunit" Version="$(FsCheckVersion)" />
```
Comment (stephentoub) [inline src/libraries/System.Collections/tests/System.Collections.Tests.csproj:137]: Do we really need this?
cc: @ViktorHofer

File: src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs (hunk 4 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs:18]: It's effectively a theory that gets fed pseudo-randomized inputs. Essentially a structured fuzzer, without being coverage guided.

Comment (stephentoub) [reply src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs]: I would prefer we not use this in any of our functional tests. If we want a dedicated fuzziness site that runs separately,  fine.

It's yet one more style of test devs need to understand.  It doesn't have the various workarounds devs know to use of something needs to be disabled, e.g. conditional. It's hard to reason about exactly what this is testing,  predictability,  and execution time. It adds another dependency to the build.  Etc.

Comment (eiriktsarpalis) [reply src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs]: I've replaced the Property attributes with regular theories that manually generate random inputs.

Comment (stephentoub) [reply src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs]: Thank you :)

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**